### ISSUE TITLE
Upgrade to use universal-router

### ISSUE THREAD

**Issue by AJolly** [azerpas/uniswap-cli#11]
Feature request: upgrade to use Uniswaps universal router.  

The new universal router lets you swap tokens/nfts, but more interestingly it allows for Permit2 - so you can execute a trade without needing to preapprove the token ahead of time. 

https://blog.uniswap.org/permit2-and-universal-router
https://github.com/Uniswap/universal-router-sdk

**Comment by azerpas** [comment]
I agree it would be a great addition! We discussed it already as shown below but I hadn't time to get onto it just yet. 
But feel free to open a PR. 😊

-----

> Sorry for the late reply!

> Effectively it seems that `UniversalRouter` [has become the preferred entrypoint for ERC20 swaps](https://docs.uniswap.org/contracts/v3/reference/deployments). 

> At first glance it would require a bit of tweaking to support it but you're very welcome to open a PR @DiMiTriFrog. 😊

_Originally posted by @azerpas in https://github.com/azerpas/uniswap-cli/issues/9#issuecomment-1710221594_
